src/NuGet.Clients/NuGet.CommandLine/Commands/InstallCommand.cs

```diff
@@ -170,7 +170,22 @@ private Task PerformV2Restore(string packagesConfigFilePath, string installPath)
                 packageRestoreFailedEvent: null,
                 sourceRepositories: packageSources.Select(sourceRepositoryProvider.CreateRepository),
                 maxNumberOfParallelTasks: DisableParallelProcessing ? 1 : PackageManagementConstants.DefaultMaxDegreeOfParallelism);
-            return PackageRestoreManager.RestoreMissingPackagesAsync(packageRestoreContext, new ConsoleProjectContext(Console));
+
+            var missingPackageReferences = installedPackageReferences.Where(reference =>
+                !nuGetPackageManager.PackageExistsInPackagesFolder(reference.PackageIdentity)).ToArray();
+
+            if (missingPackageReferences.Length == 0)
```

Comment:
If you take the above linq query and instead of enumerating it with .ToArray() just use .Any() you can check that at least one exists and it will stop after the first one instead of checking for every package on disk.